make test{-full,} fail to connect to db

Running `make test` and `make test-full` from the tip of main I eventually get this error:
```
$ make test-full 
docker-compose -f docker-compose.explorer.yml stop hez-postgres && docker-compose -f docker-compose.explorer.yml rm -f hez-postgres || true
No stopped containers
docker-compose -f docker-compose.explorer.yml up -d hez-postgres
Creating hez-postgres ... done
MallocNanoZone=0 go test -race -p 1 -timeout 600s ./...
?   	github.com/hermeznetwork/hermez-core/aggregator	[no test files]
?   	github.com/hermeznetwork/hermez-core/cmd	[no test files]
?   	github.com/hermeznetwork/hermez-core/config	[no test files]
?   	github.com/hermeznetwork/hermez-core/db	[no test files]
?   	github.com/hermeznetwork/hermez-core/encoding	[no test files]
ok  	github.com/hermeznetwork/hermez-core/etherman	(cached)
?   	github.com/hermeznetwork/hermez-core/etherman/smartcontracts/bridge	[no test files]
?   	github.com/hermeznetwork/hermez-core/etherman/smartcontracts/matic	[no test files]
?   	github.com/hermeznetwork/hermez-core/etherman/smartcontracts/proofofefficiency	[no test files]
?   	github.com/hermeznetwork/hermez-core/hex	[no test files]
?   	github.com/hermeznetwork/hermez-core/jsonrpc	[no test files]
ok  	github.com/hermeznetwork/hermez-core/log	(cached)
ok  	github.com/hermeznetwork/hermez-core/pool	(cached)
?   	github.com/hermeznetwork/hermez-core/proverclient	[no test files]
?   	github.com/hermeznetwork/hermez-core/sequencer	[no test files]
?   	github.com/hermeznetwork/hermez-core/sequencer/strategy	[no test files]
?   	github.com/hermeznetwork/hermez-core/sequencer/strategy/txprofitabilitychecker	[no test files]
?   	github.com/hermeznetwork/hermez-core/sequencer/strategy/txselector	[no test files]
ok  	github.com/hermeznetwork/hermez-core/state	(cached)
?   	github.com/hermeznetwork/hermez-core/state/pgstatestorage	[no test files]
?   	github.com/hermeznetwork/hermez-core/state/runtime	[no test files]
?   	github.com/hermeznetwork/hermez-core/state/runtime/evm	[no test files]
ok  	github.com/hermeznetwork/hermez-core/state/tree	(cached)
?   	github.com/hermeznetwork/hermez-core/synchronizer	[no test files]
?   	github.com/hermeznetwork/hermez-core/test/dbutils	[no test files]
make[1]: Entering directory '/home/fgimenez/workspace/polygon-hermez/hermez-core'
docker-compose -f docker-compose.explorer.yml stop hez-core && docker-compose -f docker-compose.explorer.yml rm -f hez-core
No stopped containers
make[1]: Leaving directory '/home/fgimenez/workspace/polygon-hermez/hermez-core'
make[1]: Entering directory '/home/fgimenez/workspace/polygon-hermez/hermez-core'
docker-compose -f docker-compose.explorer.yml stop hez-prover && docker-compose -f docker-compose.explorer.yml rm -f hez-prover
No stopped containers
make[1]: Leaving directory '/home/fgimenez/workspace/polygon-hermez/hermez-core'
make[1]: Entering directory '/home/fgimenez/workspace/polygon-hermez/hermez-core'
docker-compose -f docker-compose.explorer.yml stop hez-network && docker-compose -f docker-compose.explorer.yml rm -f hez-network
No stopped containers
make[1]: Leaving directory '/home/fgimenez/workspace/polygon-hermez/hermez-core'
--- FAIL: TestStateTransition (3.15s)
    --- FAIL: TestStateTransition/5_accounts._2_valid_tx,_3_invalid_tx_(same_amount_as_balance,_invalid_nonce,_invalid_chain_id_0) (0.00s)
        e2e_test.go:64: 
            	Error Trace:	e2e_test.go:64
            	Error:      	Received unexpected error:
            	            	failed to connect to `host=localhost user=test_user database=test_db`: failed to receive message (read tcp 127.0.0.1:42256->127.0.0.1:5432: read: connection reset by peer)
            	Test:       	TestStateTransition/5_accounts._2_valid_tx,_3_invalid_tx_(same_amount_as_balance,_invalid_nonce,_invalid_chain_id_0)
    --- FAIL: TestStateTransition/2_accounts_and_1_invalid_tx_(tx_with_more_value_than_balance)._Old_root_equals_new_root. (0.00s)
        e2e_test.go:64: 
            	Error Trace:	e2e_test.go:64
            	Error:      	Received unexpected error:
            	            	failed to connect to `host=localhost user=test_user database=test_db`: failed to receive message (read tcp 127.0.0.1:42262->127.0.0.1:5432: read: connection reset by peer)
            	Test:       	TestStateTransition/2_accounts_and_1_invalid_tx_(tx_with_more_value_than_balance)._Old_root_equals_new_root.
    --- FAIL: TestStateTransition/2_accounts_and_4_invalid_transactions._Two_of_the_transactions_have_invalid_from_and_to_address_so_no_rawTx_is_provided_for_them,_can_be_ignored.Tx_with_same_amount_than_balance_and_tx_with_invalid_chain_id_(different_chain_id_than_sequencer)._Old_root_equals_new_root (0.00s)
        e2e_test.go:64: 
            	Error Trace:	e2e_test.go:64
            	Error:      	Received unexpected error:
            	            	failed to connect to `host=localhost user=test_user database=test_db`: failed to receive message (read tcp 127.0.0.1:42268->127.0.0.1:5432: read: connection reset by peer)
            	Test:       	TestStateTransition/2_accounts_and_4_invalid_transactions._Two_of_the_transactions_have_invalid_from_and_to_address_so_no_rawTx_is_provided_for_them,_can_be_ignored.Tx_with_same_amount_than_balance_and_tx_with_invalid_chain_id_(different_chain_id_than_sequencer)._Old_root_equals_new_root
    --- FAIL: TestStateTransition/2_accounts_and_1_invalid_transaction._Wrong_encode_of_the_tx_(invalid_signature)._Old_root_equals_new_root (0.00s)
        e2e_test.go:64: 
            	Error Trace:	e2e_test.go:64
            	Error:      	Received unexpected error:
            	            	failed to connect to `host=localhost user=test_user database=test_db`: failed to receive message (read tcp 127.0.0.1:42276->127.0.0.1:5432: read: connection reset by peer)
            	Test:       	TestStateTransition/2_accounts_and_1_invalid_transaction._Wrong_encode_of_the_tx_(invalid_signature)._Old_root_equals_new_root
FAIL
FAIL	github.com/hermeznetwork/hermez-core/test/e2e	3.449s
?   	github.com/hermeznetwork/hermez-core/test/vectors	[no test files]
FAIL
make: *** [Makefile:52: test-full] Error 1
``` 
We might need to either wait a bit after creating it or properly check that it is accepting connections.

--- a/Makefile
+++ b/Makefile
@@ -41,13 +41,13 @@ build-docker: ## Build a docker image with the core binary
 .PHONY: test
 test: ## runs only short tests without checking race conditions
 	$(STOPDB) || true
-	$(RUNDB)
+	$(RUNDB); sleep 5
 	trap '$(STOPDB)' EXIT; go test -short -p 1 ./...
 
 .PHONY: test-full
 test-full: ## runs all tests checking race conditions
 	$(STOPDB) || true
-	$(RUNDB)
+	$(RUNDB); sleep 5
 	trap '$(STOPDB)' EXIT; MallocNanoZone=0 go test -race -p 1 -timeout 600s ./...
 
 .PHONY: install-linter

